api/interaction.go

```diff
@@ -23,42 +26,229 @@ const (
 	UpdateMessage
 )
 
+// InteractionResponseFlags implements flags for an
+// InteractionApplicationCommandCallbackData.
+//
+// https://discord.com/developers/docs/interactions/slash-commands#interaction-response-object-interaction-application-command-callback-data-flags
+type InteractionResponseFlags uint
+
+const EphemeralResponse InteractionResponseFlags = 64
+
 type InteractionResponse struct {
 	Type InteractionResponseType  `json:"type"`
 	Data *InteractionResponseData `json:"data,omitempty"`
 }
 
-// InteractionResponseFlags implements flags for an InteractionApplicationCommandCallbackData.
-// https://discord.com/developers/docs/interactions/slash-commands#interaction-response-interactionapplicationcommandcallbackdata
-type InteractionResponseFlags uint
+// NeedsMultipart returns true if the InteractionResponse has files.
+func (resp InteractionResponse) NeedsMultipart() bool {
+	return resp.Data != nil && resp.Data.NeedsMultipart()
+}
 
-const EphemeralResponse InteractionResponseFlags = 64
+func (resp InteractionResponse) WriteMultipart(body *multipart.Writer) error {
+	return sendpart.Write(body, resp, resp.Data.Files)
+}
 
 // InteractionResponseData is InteractionApplicationCommandCallbackData in the
 // official documentation.
 type InteractionResponseData struct {
-	TTS             option.NullableBool      `json:"tts,omitempty"`
-	Content         option.NullableString    `json:"content,omitempty"`
-	Components      *[]discord.Component     `json:"components,omitempty"`
-	Embeds          *[]discord.Embed         `json:"embeds,omitempty"`
-	AllowedMentions *AllowedMentions         `json:"allowed_mentions,omitempty"`
-	Flags           InteractionResponseFlags `json:"flags,omitempty"`
-	Files           []sendpart.File          `json:"-"`
+	// Content are the message contents (up to 2000 characters).
+	//
+	// Required: one of content, file, embeds
+	Content string `json:"content,omitempty"`
```

Comment:
I'm not sure if changing from `optional.NullableString` to `string` works here. If I remember correctly, it was nullable to null out an existing response's message content.